FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,30 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard let self = self else { return }
+
+			switch result {
+			case let .success((data, response)):
+				guard response.statusCode == 200 else {
+					return completion(.failure(Error.invalidData))
+				}
+
+				do {
+					let feedImages = try self.feedImages(from: data)
+					completion(.success(feedImages))
+				} catch {
+					completion(.failure(Error.invalidData))
+				}
```

Comment:
Following from my previous comment, by having the `map` function `throw` the correct error we don't have to overwrite it here. That way we could use a `do/catch` statement and complete with the `error` caught.
```swift
	do {
		let feedImages = try self.feedImages(from: data)
		completion(.success(feedImages))
	} catch {
		completion(.failure(error))
	}
```

Or we could use `Result`'s [initializer](https://developer.apple.com/documentation/swift/result/3139399-init) that takes a throwing closure as parameter and returns `.success` with the value returned, or `.failure` with any thrown error:
```suggestion
				completion(Result { try self.feedImages(from: data) })
```